drivers/i2c/i2c_ll_stm32.c

```diff
@@ -151,11 +151,12 @@ static int i2c_stm32_init(struct device *dev)
 		break;
 #endif /* CONFIG_I2C_1 */
 
-#ifdef CONFIG_I2C_2
+	/* I2C2 on STM32F0 uses APB1 clock as I2C clock source */
```

Comment:
This whole section feels a bit weird now (which is mainly due to HW constraints)
Can you merge this comment with comment @ line 143?